motion_control/motion_control_common/TrapezoidalProfile.cpp

```diff
@@ -0,0 +1,276 @@
+// Copyright (C) 2025 COGIP Robotics association <cogip35@gmail.com>
+// This file is subject to the terms and conditions of the GNU Lesser
+// General Public License v2.1. See the file LICENSE in the top level
+// directory for more details.
+
+/// @ingroup    motion_control_common
+/// @{
+/// @file
+/// @brief      Trapezoidal velocity profile implementation
+/// @author     Gilles Doffe <g.doffe@gmail.com>
+
+#include "motion_control_common/TrapezoidalProfile.hpp"
+
+#include <cmath>
+
+#define ENABLE_DEBUG 0
+#include <debug.h>
+
+#include <etl/algorithm.h>
+
+namespace cogip {
+
+namespace motion_control {
+
+TrapezoidalProfile::TrapezoidalProfile()
+    : initial_velocity_(0.0f), plateau_velocity_(0.0f), target_distance_(0.0f),
+      final_velocity_(0.0f), accel_periods_(0), plateau_periods_(0), decel_periods_(0),
+      acceleration_(0.0f), deceleration_(0.0f), initialized_(false)
+{
+}
+
+float TrapezoidalProfile::compute_accel_distance(float v0, float v1, float accel)
+{
+    // Distance = (v1² - v0²) / (2 * a)
+    // Handle division by zero
+    if (etl::absolute(accel) < 1e-6f) {
+        return 0.0f;
+    }
+    return (v1 * v1 - v0 * v0) / (2.0f * accel);
+}
+
+uint32_t TrapezoidalProfile::compute_accel_periods(float v0, float v1, float accel)
+{
+    // Number of periods = |v1 - v0| / a
+    // Handle division by zero
+    if (etl::absolute(accel) < 1e-6f) {
+        return 0;
+    }
+    float periods = etl::absolute(v1 - v0) / accel;
+    // Round to nearest integer: add 0.5 and truncate
+    return static_cast<uint32_t>(periods + 0.5f);
+}
+
+uint32_t TrapezoidalProfile::generate_optimal_profile(float initial_velocity, float target_distance,
+                                                      float acceleration, float deceleration,
+                                                      float max_speed, bool must_stop_at_end)
+{
+    // Reset state
+    initialized_ = false;
+
+    // Validate inputs
+    if (etl::absolute(target_distance) < 1e-6f) {
+        // No distance to travel
+        initial_velocity_ = initial_velocity;
+        plateau_velocity_ = initial_velocity;
+        target_distance_ = 0.0f;
+        final_velocity_ = initial_velocity;
+        accel_periods_ = 0;
+        plateau_periods_ = 0;
+        decel_periods_ = 0;
+        acceleration_ = 0.0f;
+        deceleration_ = 0.0f;
+        initialized_ = true;
+        return 0;
+    }
+
+    if (acceleration <= 0.0f || deceleration <= 0.0f || max_speed <= 0.0f) {
+        return 0; // Invalid parameters
```

Comment:
Maybe we should also ensure now distance to travel as above here also, WDYT ? 

It yes, maybe we can move:
```c++
// No distance to travel
        initial_velocity_ = initial_velocity;
        plateau_velocity_ = initial_velocity;
        target_distance_ = 0.0f;
        final_velocity_ = initial_velocity;
        accel_periods_ = 0;
        plateau_periods_ = 0;
        decel_periods_ = 0;
        acceleration_ = 0.0f;
        deceleration_ = 0.0f;
        initialized_ = true;
```

to something like a reset method to be reused on error cases.